iio/iiod.c

```diff
@@ -0,0 +1,814 @@
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include <inttypes.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+#include "iiod.h"
+#include "iiod_private.h"
+
+#include "no-os/error.h"
+#include "no-os/util.h"
+
+#define SET_DUMMY_IF_NULL(func, dummy) ((func) ? (func) : (dummy))
+
+static char delim[] = " \r\n";
+
+static struct iiod_str cmds[] = {
+	[IIOD_CMD_HELP]		= IIOD_STR("HELP"),
+	[IIOD_CMD_EXIT]		= IIOD_STR("EXIT"),
+	[IIOD_CMD_PRINT]	= IIOD_STR("PRINT"),
+	[IIOD_CMD_VERSION]	= IIOD_STR("VERSION"),
+	[IIOD_CMD_TIMEOUT]	= IIOD_STR("TIMEOUT"),
+	[IIOD_CMD_OPEN]		= IIOD_STR("OPEN"),
+	[IIOD_CMD_CLOSE]	= IIOD_STR("CLOSE"),
+	[IIOD_CMD_READ]		= IIOD_STR("READ"),
+	[IIOD_CMD_WRITE]	= IIOD_STR("WRITE"),
+	[IIOD_CMD_READBUF]	= IIOD_STR("READBUF"),
+	[IIOD_CMD_WRITEBUF]	= IIOD_STR("WRITEBUF"),
+	[IIOD_CMD_GETTRIG]	= IIOD_STR("GETTRIG"),
+	[IIOD_CMD_SETTRIG]	= IIOD_STR("SETTRIG"),
+	[IIOD_CMD_SET]		= IIOD_STR("SET")
+};
+static const uint32_t priority_array[ARRAY_SIZE(cmds)] = {
```

Comment:
The size of the array will be derived from the content. If you want to make sure both arrays are in sync, I'd suggest to use a `static_assert(ARRAY_SIZE(cmds) == ARRAY_SIZE(priority_array), "Arrays must have the same size");`